code/modules/mob/living/carbon/xenomorph/castes/hivemind/abilities_hivemind.dm

```diff
@@ -1 +1,177 @@
-// empty file for now
+// ***************************************
+// *********** Reposition Core
+// ***************************************
+/datum/action/xeno_action/activable/reposition_core
+	name = "Reposition Core"
+	action_icon_state = "emit_neurogas"
+	mechanics_text = "Reposition your core to a target location on weeds. The further away the location is the longer this will take up to a maximum of 60 seconds. 3 minute cooldown."
+	ability_name = "reposition core"
+	plasma_cost = 150
+	cooldown_timer = 180 SECONDS
+	keybind_flags = XACT_KEYBIND_USE_ABILITY
+	keybind_signal = COMSIG_XENOABILITY_REPOSITION_CORE
+
+/datum/action/xeno_action/activable/reposition_core/on_cooldown_finish()
+	owner.playsound_local(owner.loc, 'sound/voice/alien_drool1.ogg', 50, TRUE)
+	to_chat(owner, "<span class='xenonotice'>We regain the strength to reposition our neural core.</span>")
+	return ..()
+
+/datum/action/xeno_action/activable/reposition_core/use_ability(atom/A)
+	var/mob/living/carbon/xenomorph/hivemind/X = owner
+	var/obj/effect/alien/hivemindcore/core = X.core
+	var/turf/target_turf = get_turf(X)
+	var/distance = get_dist(target_turf, get_turf(core))
+
+	var/delay = max(HIVEMIND_REPOSITION_CORE_DELAY_MIN, HIVEMIND_REPOSITION_CORE_DELAY_MOD * distance) //Calculate the distance scaling delay before we complete the reposition
+
+
+	if(!check_build_location(target_turf, X)) //Check target turf for suitability
```

Comment:
this should likely be moved into can_use_ability()

Reply:
As mentioned on discord, I got some wonky looping behaviour with can_use_ability, but I'll try again and see if it can't be coerced into working properly

Reply:
This is done.